static/app/components/dropdownMenuV2.tsx

```diff
@@ -0,0 +1,280 @@
+import {Fragment, useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+import {FocusScope} from '@react-aria/focus';
+import {useKeyboard} from '@react-aria/interactions';
+import {AriaMenuOptions, useMenu} from '@react-aria/menu';
+import {
+  AriaPositionProps,
+  OverlayProps,
+  PositionAria,
+  useOverlay,
+  useOverlayPosition,
+} from '@react-aria/overlays';
+import {useSeparator} from '@react-aria/separator';
+import {mergeProps} from '@react-aria/utils';
+import {useTreeState} from '@react-stately/tree';
+import {Node} from '@react-types/shared';
+
+import MenuControl from 'sentry/components/dropdownMenuControlV2';
+import MenuItem, {MenuItemProps} from 'sentry/components/dropdownMenuItemV2';
+import MenuSection from 'sentry/components/dropdownMenuSectionV2';
+import space from 'sentry/styles/space';
+
+type Props = {
+  onClose?: () => void;
+  /**
+   * Whether the menu should close when an item has been clicked/selected
+   */
+  closeOnSelect?: boolean;
+  /**
+   * Ref object to the trigger element, needed for useOverlayPosition()
+   */
+  triggerRef: React.RefObject<HTMLButtonElement>;
+  /**
+   * Whether this is a submenu
+   */
+  isSubmenu: boolean;
+  /*
+   * Title to display on top of the menu
+   */
+  menuTitle?: string;
+  /**
+   * If this is a submenu, it will in some cases need to close the root menu
+   * (e.g. when a submenu item is clicked).
+   */
+  closeRootMenu: () => void;
+  /**
+   * If this is a submenu, it will in some cases need to close itself (e.g.
+   * when the user presses the arrow left key)
+   */
+  closeCurrentSubmenu?: () => void;
+} & AriaMenuOptions<MenuItemProps> &
+  Partial<OverlayProps> &
+  Partial<AriaPositionProps>;
```

Comment:
```suggestion
type Props = {
  /**
   * Ref object to the trigger element, needed for useOverlayPosition()
   */
  triggerRef: React.RefObject<HTMLButtonElement>;
  /**
   * Whether this is a submenu
   */
  isSubmenu: boolean;
  /**
   * Whether the menu should close when an item has been clicked/selected
   */
  closeOnSelect?: boolean;
  onClose?: () => void;
  /*
   * Title to display on top of the menu
   */
  menuTitle?: string;
  /**
   * If this is a submenu, it will in some cases need to close the root menu
   * (e.g. when a submenu item is clicked).
   */
  closeRootMenu: () => void;
  /**
   * If this is a submenu, it will in some cases need to close itself (e.g.
   * when the user presses the arrow left key)
   */
  closeCurrentSubmenu?: () => void;
} & AriaMenuOptions<MenuItemProps> &
  Partial<OverlayProps> &
  Partial<AriaPositionProps>;
```